backend/controllers/animals.js

```diff
@@ -38,8 +38,23 @@ const filterByNameContainsValue = (req, res, next) => {
     .catch(error => next(error));
 };
 
+const filterByOrder = (req, res, next) => {
```

Comment:
I think you don't need separate method for filtering and search. Just support query parameters in animalsList method